### PR TITLE
LONDON CLASS 8 - BERKELI HALMYRADOV - JS2 - WEEK1

### PR DESCRIPTION
**Pull request by berkeli**
**Volunteers: Are you marking this coursework?** _You can find a guide on how to mark this coursework in `HOW_TO_MARK.md` in the root of this repository_

# Your Details

- Your Name: Berkeli Halmyradov
- Your City: London
- Your Slack Name: Berkeli

# Homework Details

- Module: JS-CORE 2
- Week: 1

# Notes

- What did you find easy?

- What did you find hard?

- What do you still not understand?

- Any other notes?

### CODE REVIEW

**Comment by berkeli** [comment]
Hi @gixo, 

This week's homework

File: 2-mandatory/1-recipes.js [hunk 1]
```diff
@@ -22,4 +22,37 @@
...
+  serves: 2,
+  ingredients: ["Flour", "Sugar", "Milk", "Chocolate", "Butter"],
+};
+
+const recipes = [recipe1, recipe2, recipe3, recipe4, recipe5];
+
+recipes.forEach((e, i) => {
+  console.log(`%c#${i}. Recipe for: ${e.title}`, "color: red; font-weight: 800; font-size: 1.5rem");
+  console.log(`Serves: ${e.serves} ${e.serves > 1 ? "people" : "person"}`);
+  console.log("%cIngredients: ", "color: green; font-weight: 800; font-size: 1.5rem");
```
Comment (gixo) [inline 2-mandatory/1-recipes.js:56]: Love the custom Javascript styling here.

File: 2-mandatory/2-currency-code-lookup.js [hunk 2]
```diff
@@ -17,21 +18,36 @@ const COUNTRY_CURRENCY_CODES = [
...
+const createLookup = (countryCurrencyCodes) => {
+  const answer = {};
+  for (const el of countryCurrencyCodes) {
+    const [country, currency] = el;
+    answer[country] = currency;
+  }
+  return answer;
+};
+
+const createLookup2 = (countryCurrencyCodes) => Object.fromEntries(countryCurrencyCodes);
```
Comment (gixo) [inline 2-mandatory/2-currency-code-lookup.js:30]: Nice way to resolve the exercise with two different approaches 👍

File: 3-extra/1-count-words.js [hunk 3]
```diff
@@ -25,8 +26,9 @@
 
 function countWords(string) {
   const wordCount = {};
-
-  // write code here
+  if (!string) return wordCount;
+  // split into array // count words, word exists add 1 otherwise it's equal to 1
+  string.split(" ").forEach((e) => (wordCount[e] ? wordCount[e]++ : wordCount[e] = 1));
```
Comment (gixo) [inline 3-extra/1-count-words.js:31]: Nice advanced use of `forEach` function with the conditional ternary operator 👍

**Review by gixo — APPROVED** [review]
Well done Berkeli, this is another solid submission. You have demonstrated great knowledge of the most used Javarous array methods, combined with arrow functions and object manipulation. Way to go!

As usual, keep yourself challenged and keep learning by completing increasingly difficult algorithms problems, you are doing excellently so far!

**Pull request closed without merging**